src/Data/Nat/Properties.agda

```diff
@@ -1747,6 +1751,80 @@ m≤n+∣m-n∣ m n = subst (m ≤_) (cong (n +_) (∣-∣-comm n m)) (m≤n+∣
 m≤∣m-n∣+n : ∀ m n → m ≤ ∣ m - n ∣ + n
 m≤∣m-n∣+n m n = subst (m ≤_) (+-comm n _) (m≤n+∣m-n∣ m n)
 
+∣-∣-triangle : TriangleInequality ∣_-_∣
+∣-∣-triangle zero    y       z       = m≤n+∣n-m∣ z y
+∣-∣-triangle x       zero    z       = begin
+  ∣ x - z ∣     ≤⟨ ∣m-n∣≤m⊔n x z ⟩
+  x ⊔ z         ≤⟨ m⊔n≤m+n x z ⟩
+  x + z         ≡⟨ cong₂ _+_ (sym (∣-∣-identityʳ x)) refl ⟩
+  ∣ x - 0 ∣ + z ∎
+  where open ≤-Reasoning
+∣-∣-triangle x       y       zero    = begin
+  ∣ x - 0 ∣             ≡⟨ ∣-∣-identityʳ x ⟩
+  x                     ≤⟨ m≤∣m-n∣+n x y ⟩
+  ∣ x - y ∣ + y         ≡⟨ cong₂ _+_ refl (sym (∣-∣-identityʳ y)) ⟩
+  ∣ x - y ∣ + ∣ y - 0 ∣ ∎
+  where open ≤-Reasoning
+∣-∣-triangle (suc x) (suc y) (suc z) = ∣-∣-triangle x y z
+
+------------------------------------------------------------------------
+-- Metric structures
```

Comment:
I think this is a bad place to put these properties - Nats have so many properties, if they all end up in one, eventually huge, file, things will devolve.  Perhaps `Data.Nat.Metric.Properties`?

Reply:
Hmm so I agree that ideally we'd have a `Data.Nat.Metric` subdirectory, just like we have a `Data.List.Relation`. However I'd want it to be done consistently, e.g. move all the properties of `|_-_|` out of `Data.Nat.Properties` at the same time. It may be difficult to do backwards compatibly(?) so I think it's probably best done when we get around to #619.